### PR TITLE
docs(Documentation): Update User Guide to Support June 2023 Release

### PR DESCRIPTION
**Pull request by Jake-Carter**
### Description

This PR updates the MSDK User Guide.  A preview of the new version can be viewed here:  https://jake-carter.github.io/msdk/USERGUIDE/

Major changes:
* Add support for the [details](https://facelessuser.github.io/pymdown-extensions/extensions/details/) PyMarkdown extension.  The User Guide is now populated with helpful notes/warning dropdowns with improved formatting.
image

* Add documentation for headless/command-line installations ([preview](https://jake-carter.github.io/msdk/USERGUIDE/#command-line-installation))

* General formatting improvements with the addition of custom CSS
  * Code blocks are now surrounded by an outline:
  image
  * Tables (mostly) no longer stretch across the boundary of the page.
  image

* Update Tables for Build Config Variables ([preview](https://jake-carter.github.io/msdk/USERGUIDE/#build-tables))
  * Missing variables have been documented
  * Table have been further organized into sub-sections
  
* Add documentation on VS Code Disassembly View ([preview](https://jake-carter.github.io/msdk/USERGUIDE/#disassembly-view))

* Add `setenv.sh` as an option for Command-Line setup instructions ([preview](https://jake-carter.github.io/msdk/USERGUIDE/#setup-command-line))

* Document make targets for flashing (`make flash.openocd`) ([preview](https://jake-carter.github.io/msdk/USERGUIDE/#flashing-on-the-command-line))

* Fix typo in BSP Search Directory code example ([preview](https://jake-carter.github.io/msdk/USERGUIDE/#bsp-search-directory))

* Add auto-generated tables of examples for all microcontrollers ([preview](https://jake-carter.github.io/msdk/USERGUIDE/#examples))

* Add MAX32662 web links now that part is released.

* Add CoreMark library section ([preview](https://jake-carter.github.io/msdk/USERGUIDE/#coremark))

* Add Debugger documentation ([preview](https://jake-carter.github.io/msdk/USERGUIDE/#debuggers))
  * Hardware limitations of debug
  * MAX32625PICO documentation, firmware updates, and erase instructions

### Checklist Before Requesting Review

- [x] PR Title follows correct guidelines.
- [x] Description of changes and all other relevant information.
- [x] (Optional) Link any related GitHub issues [using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [x] (Optional) Provide info on any relevant functional testing/validation.  For API changes or significant features, this is not optional.

### CODE REVIEW

**Comment by Jake-Carter** [comment]
@Analog-Devices-MSDK/msdk_maintainers your review of the updated User Guide would be greatly appreciated!

**Comment by Jacob-Scheiffler** [comment]
The new examples section is a lot and very redundant since most parts support similar examples. We should consider having a list of common examples and list(s) of examples that are unique to each part, like the CNN examples for the AI parts? Or something like that? I understand that the maintenance overhead is increased by doing it that way, but it would reduce a lot of clutter.

**Comment by Jake-Carter** [comment]
> The new examples section is a lot and very redundant since most parts support similar examples. We should consider having a list of common examples and list(s) of examples that are unique to each part, like the CNN examples for the AI parts? Or something like that? I understand that the maintenance overhead is increased by doing it that way, but it would reduce a lot of clutter.
> 
> Other than that I think the new changes look good!

I think it's a good idea that we should apply to the organization of the examples themselves.  We've talked about having "Common Examples" vs "Part-specific examples" folders in the past.

Within the context of this PR I think it's a little out of scope, but I'm open to removing the tables until we reorganize

**Comment by Jake-Carter** [comment]
Examples section has been updated with "Common Examples" vs part-specific examples [(preview)](https://jake-carter.github.io/msdk/USERGUIDE/#examples)

File: Examples/MAX32662/Demo/main.c [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
 /**
  * @file    main.c
- * @brief   Demo
+ * @brief   This example demonstrates the use of the TFT Display, UART Terminal, RTC, Pushbuttons, and LEDs on the MAX32662EVKIT.
```
Comment (sihyung-maxim) [inline Examples/MAX32662/Demo/main.c:36]: MAX32662EVKIT gets added in the common examples.
image

**Comment by sihyung-maxim** [comment]
Should we leave the ME55 in the common examples? These examples specifically are untested and must be updated when we start the bring-up.
image

**Review by Jacob-Scheiffler — APPROVED** [review]

File: Examples/MAX32662/Demo/main.c (hunk 1 above)
Comment (Jake-Carter) [reply Examples/MAX32662/Demo/main.c]: Fixed https://github.com/Analog-Devices-MSDK/msdk/pull/683/commits/83fdfed973e788133e73a8ba476ac13e159964f7

**Comment by Jake-Carter** [comment]
> Should we leave the ME55 in the common examples? These examples specifically are untested and must be updated when we start the bring-up. image

Thanks, blacklisted the MAX32572.  https://github.com/Analog-Devices-MSDK/msdk/pull/683/commits/66c25f79a210d562fc0bafa34ccd49063f899a63

**Review by sihyung-maxim — APPROVED** [review]

**Pull request merged**